### PR TITLE
Progress removing //base dependency

### PR DESCRIPTION
**Pull request by abarth**
Split into several commits for reviewability.

### CODE REVIEW

**Comment by abarth** [comment]
Depends on https://fuchsia-review.googlesource.com/c/5049/

File: flutter/lib/ui/painting/image_decoding.cc [hunk 1]
```diff
@@ -130,12 +129,12 @@ void DecodeImageFromList(Dart_NativeArguments args) {
   PassOwnPtr<DartPersistentValue> callback =
       adoptPtr(new DartPersistentValue(DartState::Current(), callback_handle));
 
-  RefPtr<SharedBuffer> buffer = SharedBuffer::create();
-  buffer->append(reinterpret_cast<const char*>(list.data()),
-                 list.num_elements());
+  const char* bytes = reinterpret_cast<const char*>(list.data());
+  PassOwnPtr<std::vector<char>> buffer =
```
Comment (chinmaygarde) [inline flutter/lib/ui/painting/image_decoding.cc]: I think a dedicated class that manages resizable allocations would serve us well. `vector<char>` is an awkward substitute.

Comment (chinmaygarde) [inline flutter/lib/ui/painting/image_decoding.cc]: Generalized lambda capture would avoid the need for the pointer right? Are we switching to C++14? I thought I saw some GN files with that flag.

**Comment by chinmaygarde** [comment]
lgtm

File: flutter/lib/ui/painting/image_decoding.cc (hunk 1 above)
Comment (abarth) [inline flutter/lib/ui/painting/image_decoding.cc]: > I think a dedicated class that manages resizable allocations would serve us well.

Comment (abarth) [inline flutter/lib/ui/painting/image_decoding.cc]: > I think a dedicated class that manages resizable allocations would serve us well.

Yeah, you're probably right.

> Generalized lambda capture would avoid the need for the pointer right?

That's right.  A lot of this code becomes prettier with generalized lambda capture.

> Are we switching to C++14? I thought I saw some GN files with that flag.

Fuchsia is on C++14.  The tricky part about switching is getting a toolchain for Android that supports C++14.  It's possible if we upgrade our NDK and bring our own libc++ that we could get C++14 on Android.

We don't have too much of this stuff, so I've just been working around the lack of generalized lambda capture so far.  If it gets too painful, we can try to get a C++14 toolchain going for Android.

**Comment by abarth** [comment]
Pushed manually to preserve history and avoid merge commit.

**Pull request closed without merging**